### PR TITLE
Capsule provisioning

### PR DESCRIPTION
**Pull request by elyezer**
Add setup_capsule_firewall task and update the setup_firewall to
setup_satellite_firewall. The setup_firewall task now is a helper for
setting up firewall rules and using iptables on RHEL6 and firewall-cmd
on RHEL7.

Add setup_capsule_content task to help setting up a custom content on
the Satellite Server that allow creating a capsule.

Update capsule_installer to match the new way to run the capsule
installer.

Update setup_capsules to not start content sync. This may be added later
if syncing is something really needed.

### CODE REVIEW

File: automation_tools/satellite6/capsule.py [hunk 1]
```diff
@@ -393,7 +611,7 @@ def setup_capsules(path):
         with settings(host_string=capsule.host_string):
             register_capsule()
             put(local_path=cert_path)
-            capsule_installer(*oauth_info)
+            capsule_installer(capsule.hostname, cert_path, *oauth_info)
 
     # Start content synchronization in all registered capsules
```
Comment (kbidarkar) [inline automation_tools/satellite6/capsule.py]: We can now do syncing from the WEBUI too, so I feel we should have a arg/flag to enable ot disable it.

**Comment by sthirugn** [comment]
It is sad to see that hammer commands need to be rewritten here.  I understand that the intent here is not to make automation-tools dependent on nailgun or other tools.  But since we are using hammer, we may need to take care of its changes according to satellite versions, if they happen to differ in future due to some reason.  Is there an other easier alternative - like using `nailgun`?  perhaps keeping it as an optional dependency and use it only when installing capsule? just thinking out loud...

File: automation_tools/satellite6/capsule.py [hunk 2]
```diff
@@ -225,7 +217,8 @@ def register_capsule():
     """Register the capsule on the Satellite 6 server."""
     config = _get_config()
     run(
-        'rpm -Uvh http://{}/pub/katello-ca-consumer-latest.noarch.rpm'
+        'yum -y localinstall '
+        'http://{}/pub/katello-ca-consumer-latest.noarch.rpm'
```
Comment (sthirugn) [inline automation_tools/satellite6/capsule.py]: just curious, what is the need for this change?

**Comment by sthirugn** [comment]
ACK pending question.  Note: my question is not a blocker.

File: automation_tools/satellite6/capsule.py (hunk 2 above)
Comment (elyezer) [inline automation_tools/satellite6/capsule.py]: It is needed since this is the way the UI page recommends customers to install the certificates RPM.

Comment (sthirugn) [inline automation_tools/satellite6/capsule.py]: Which UI page you mean? I see latest Hosts -> Content Hosts page still refers to rpm command.

Comment (elyezer) [inline automation_tools/satellite6/capsule.py]: @sthirugn sorry, it is from the output from the command you use to generate the certificates `capsule-certs-generate`.

Comment (sthirugn) [inline automation_tools/satellite6/capsule.py]: @elyezer no worries, ty for the explanation :+1:

**Comment by elyezer** [comment]
@kbidarkar @sthirugn if you are ok with this PR, can you please merge it?

**Comment by elyezer** [comment]
Rebased

**Comment by sthirugn** [comment]
This already has my ACK.  Can somebody else do a second ack/merge please?

File: automation_tools/__init__.py [hunk 3]
```diff
@@ -353,63 +352,124 @@ def setup_fake_manifest_certificate(certificate_url=None):
...
+                'udp': (
+                    4242,
+                    4422,
+                ),
+            }
+    """
+    if definitions is None:
+        return
+
+    if distro_info()[1] >= 7:
```
Comment (omaciel) [inline automation_tools/__init__.py]: Should this be `6` instead of `7`?

File: automation_tools/satellite6/capsule.py (hunk 2 above)
Comment (omaciel) [inline automation_tools/satellite6/capsule.py]: I thought `localinstall` was needed only if the package was local to the filesystem where the command is being run?

**Comment by omaciel** [comment]
@elyezer couple of comments

File: automation_tools/__init__.py (hunk 3 above)
Comment (elyezer) [inline automation_tools/__init__.py]: Good catch, this actually needs to be < 7 or == 6

Comment (elyezer) [inline automation_tools/__init__.py]: Like here https://github.com/SatelliteQE/automation-tools/pull/308/files#diff-a692ae115cc45cc7627d4545e9b4c294R394

File: automation_tools/satellite6/capsule.py (hunk 2 above)
Comment (elyezer) [inline automation_tools/satellite6/capsule.py]: localinstall have the same effect as rpm -Uvh. When  passing an URL it's gonna first download the RPM and then install it locally

**Comment by elyezer** [comment]
Comments addressed. Thanks @omaciel for the eagle eyes.

**Comment by omaciel** [comment]
ACK

**Pull request merged**